plonky2x/src/frontend/eth/beacon/builder.rs

```diff
@@ -40,7 +45,47 @@ impl<F: RichField + Extendable<D>, const D: usize> CircuitBuilder<F, D> {
         self.add_simple_generator(&generator);
         generator.validator
     }
+
+    /// SHA256 implementation for builder 
+        pub fn sha(&mut self, input: &[ByteVariable]) -> Bytes32Variable {
+            let input_bool: Vec<BoolTarget> = input.iter().flat_map(|byte| byte.0.iter().cloned().map(|bool_variable| BoolTarget::new_unsafe(bool_variable.0 .0))).collect();
```

Comment:
nit: I think we should have a method on `ByteVariable` that just directly returns a slice of BoolTargets so that we can reuse the logic?